### DISCUSSION TITLE
Importing Grayscale RAW Images as Colour Channels

### DISCUSSION THREAD

**Discussion by AngelaDMerkel** [libvips/libvips#4082] [Help wanted]
I am looking for a way to import three different grayscale raws as each one R, G, and B colour channel. My use case is digitising negative film using alternating outputs of red, green and blue LEDs to produce a more veracious rendition of the dyes. I am trying to emulate the way older film scanners worked. 

Is what I am considering possible in vips?

**Comment by jcupitt** [comment]
Hello @AngelaDMerkel, 

Sure, just load them, then use `bandjoin` to join them together bandwise. You'll need to tag the result as rgb, or most of the savers will think you have a greyscale image with two alphas.

In python (untested):

```python
bands = [pyvips.Image.new_from_file(f"{band}.png") for band in ["r", "g", "b"]]
rgb = bands[0].bandjoin(bands[1:]).copy(interpretation="srgb")
```
